src/map/unit.cpp

```diff
@@ -149,29 +165,53 @@ bool unit_walktoxy_nextcell(block_list& bl, bool sendMove, t_tick tick) {
 	if (ud == nullptr)
 		return true;
 
-	int32 speed;
-
 	// Reached end of walkpath
 	if (ud->walkpath.path_pos >= ud->walkpath.path_len)
 		return false;
 
+	// Monsters first check for a chase skill and if they didn't use one if their target is in range each cell after checking for a chase skill
+	if (bl.type == BL_MOB) {
+		mob_data* md = reinterpret_cast<mob_data*>(&bl);
```

Comment:
```suggestion
		mob_data& md = reinterpret_cast<mob_data&>(bl);
```
And change the rest below accordingly.